app/jobs/cache_analytics_job.rb

```diff
@@ -112,152 +89,310 @@ def self.monitoree_counts_by_age_group(analytic_id, monitorees, active_monitorin
         WHEN TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) >= 50 AND TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) < 60 THEN '50-59'
         WHEN TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) >= 60 AND TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) < 70 THEN '60-69'
         WHEN TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) >= 70 AND TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) < 80 THEN '70-79'
-        WHEN TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) >= 80 THEN '>=80'
+        WHEN TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) >= 80 AND TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) < 115 THEN '>=80'
+        WHEN TIMESTAMPDIFF(YEAR, date_of_birth, CURDATE()) >= 115 THEN 'FAKE_BIRTHDATE'
       END
     SQL
-    monitorees.monitoring_active(active_monitoring)
-              .group(age_groups, :exposure_risk_assessment)
-              .order(Arel.sql(age_groups), :exposure_risk_assessment)
-              .size
-              .map do |(age_group, risk), total|
-                monitoree_count(analytic_id, active_monitoring, 'Age Group', age_group, risk, total)
-              end
+    WORKFLOWS.map do |workflow|
+      monitorees.where(isolation: workflow == 'Isolation').monitoring_active(true)
+                .group(age_groups)
+                .order(Arel.sql(age_groups))
+                .size
+                .map do |(age_group), total|
+                  counts.append(monitoree_count(analytic_id, true, 'Age Group', age_group, total, workflow))
+                end
+    end
+    counts
   end
 
   # Monitoree counts by sex
-  def self.monitoree_counts_by_sex(analytic_id, monitorees, active_monitoring)
-    monitorees.monitoring_active(active_monitoring)
-              .group(:sex, :exposure_risk_assessment)
-              .order(:sex, :exposure_risk_assessment)
-              .size
-              .map do |(sex, risk), total|
-                monitoree_count(analytic_id, active_monitoring, 'Sex', sex.nil? ? 'Missing' : sex, risk, total)
-              end
+  def self.monitoree_counts_by_sex(analytic_id, monitorees)
+    counts = []
+    WORKFLOWS.map do |workflow|
```

Comment:
although this works, maybe something we can do to shorten this code a bit is to do `.group(:sex, :isolation)` and then `.map do |(sex, isolation), total|` and pass `isolation ? 'Isolation' : 'Exposure'` as the `workflow` argument

Reply:
^ that should also reduce the number of queries by 1/2 because we're having the db do the grouping instead of us

Reply:
sorry for not suggesting this earlier, but yeah I think we should definitely apply this change to all of these methods for performance reasons